src/Plugin/Validation/Constraint/LowercaseEmailConstraintValidator.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Drupal\apigee_edge\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the LowercaseEmail constraint.
+ */
+class LowercaseEmailConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint) {
+    foreach ($value as $item) {
+      // First check if the organization is ApigeeX.
+      try {
+        /** @var \Drupal\apigee_edge\SDKConnectorInterface $sdk_connector */
+        $sdk_connector = \Drupal::service('apigee_edge.sdk_connector');
+        $org_controller = \Drupal::service('apigee_edge.controller.organization');
+        /** @var \Apigee\Edge\Api\Management\Entity\Organization $organization */
+        $organization = $org_controller->load($sdk_connector->getOrganization());
+
+        // Check if organization is ApigeeX.
+        if ($organization && ('CLOUD' === $organization->getRuntimeType() || 'HYBRID' === $organization->getRuntimeType())) {
```

Comment:
Can this check for ApigeeX instance be moved to the module file instead of here?

Reply:
What would be the benefit of that? The constraint is added in "build-time", however this should be a runtime check because what if a site gets re-connected to an Apigee X org after it was connected to Apigee Edge.

Reply:
What may help on potential performance issues caused by fetching org information multiple times (let's say when a batch operation is running) is storing the fact about whether this is an Apigee X org in an in-memory cache (or adding in-memory cache to the org controller if it does not have it yet).